agent/agents/mysql/perfschema/perfschema.go

```diff
@@ -119,6 +120,37 @@ type newPerfSchemaParams struct {
 
 const queryTag = "agent='perfschema'"
 
+// getPerfschemaSummarySize returns size of rows for perfschema summary cache.
+func getPerfschemaSummarySize(q reform.Querier, l *logrus.Entry) uint {
+	var name string
+	var size uint
+
+	query := fmt.Sprintf("SHOW VARIABLES /* %s */ LIKE 'performance_schema_digests_size'", queryTag)
+	err := q.QueryRow(query).Scan(&name, &size)
+	if err != nil {
+		size = summariesCacheSize
```

Comment:
It may be a good idea to log error, maybe with debug level. Otherwise it may become tricky to discover problems with this code.